common/unsafe/src/main/java/org/apache/spark/unsafe/hash/Murmur3_x86_32.java

```diff
@@ -52,13 +53,31 @@ public int hashUnsafeWords(Object base, long offset, int lengthInBytes) {
     return hashUnsafeWords(base, offset, lengthInBytes, seed);
   }
 
+  public static int hashUnsafeWordsBlock(MemoryBlock base, long offset, int lengthInBytes, int seed) {
+    return hashUnsafeWords(base.getBaseObject(), offset, lengthInBytes, seed);
+  }
+
   public static int hashUnsafeWords(Object base, long offset, int lengthInBytes, int seed) {
     // This is based on Guava's `Murmur32_Hasher.processRemaining(ByteBuffer)` method.
     assert (lengthInBytes % 8 == 0): "lengthInBytes must be a multiple of 8 (word-aligned)";
     int h1 = hashBytesByInt(base, offset, lengthInBytes, seed);
     return fmix(h1, lengthInBytes);
   }
 
+  public static int hashUnsafeBytesBlock(MemoryBlock base, int seed) {
+    long offset = base.getBaseOffset();
```

Comment:
It is better to leave a comment why we have this.

Reply:
Can't we just call `hashUnsafeBytes` with the block's baseObject, baseOffset and size?

Reply:
While not to call `hashUnsafeBytes`  can achieve better performance, I will call `hashUnsafeBytes ` until we will remove `hashUnsafeBytes`.

Reply:
Do we need to leave a comment at every method with `...ByteBlock(`?

Reply:
I am considering to change to call `hashUnsafeBytesBlock()` from `hashUnsafeBytes()`. It can also reduce the duplication. Does it make sense?

Reply:
Seems they are somehow duplicate.

Reply:
Sure, to eliminate duplication, we will call `hashUnsafeBytesBlock()` from `hashUnsafeBytes()`.